tests/phpunit/elementor/modules/atomic-widgets/template-renderer/test-template-renderer.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace Elementor\Testing\Modules\AtomicWidgets\TemplateRenderer;
+
+use Elementor\Modules\AtomicWidgets\Elements\Atomic_Widget_Base;
+use Elementor\Modules\AtomicWidgets\TemplateRenderer\Has_Template;
+use Elementor\Modules\AtomicWidgets\TemplateRenderer\Template_Renderer;
+use ElementorEditorTesting\Elementor_Test_Base;
+use Spatie\Snapshots\MatchesSnapshots;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Template_Renderer extends Elementor_Test_Base {
+	use MatchesSnapshots;
+
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+
+		Template_Renderer::reset();
+	}
+
+	public function set_up(): void {
+		parent::set_up();
+
+		Template_Renderer::reset();
+	}
```

Comment:
why do you need it twice?
I think the second one is good enough, no?

Reply:
Changed. the idea is to make each test clean and also clean it for other tests to use